3.0.0 changes

cs-abhishek-shukla opened the pull request:
- Added release notes and made changes in the doc as per RN
- Removed mention of various modes
- Enhanced release notes to include other features as per feedback

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -3,21 +3,31 @@
-This widget comes pre-configured and ready for usage.
 
+> [!NOTE]  
+> The **AI Assistant** launches when module records such as alerts, incidents, or indicators are accessed and does not require an independent configuration.
```
module list is not limited to above only

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -3,21 +3,31 @@
+- `Read` and `Usage` permissions on Widgets
+- `Read` and `Execute` permissions on Playbooks
+
+For The **FortiAI** bot to work, you must install and configure the **FortiAI** solution pack
```
this is a pre-requisite

cs-abhishek-shukla committed:
3.0.0 changes
- Added release notes and made changes in the doc as per RN
- Removed mention of various modes

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,93 +3,101 @@
+- Click on any of the questions, and a dialog box appears to Review and Edit the data that will be sent to the LLM like OpenAI.
+
+    >[!NOTE]
+    >All sensitive and confidential information is masked automatically by the bot. Please review it carefully and edit it before submitting it to LLM.
```
Please review it carefully and edit it before submitting it to LLM. --> this is no longer supported in bot now. please remove this text.

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,93 +3,101 @@
 # Usage
 
-The AI Assistant Widget adds the 'FortiAI' bot to both the list view and detail view of modules in FortiSOAR, as well as in the playbook designer. 'FortiAI' is a valuable addition to your security toolkit, making playbook design more efficient and insightful. Whether you are new to security automation or a seasoned professional, 'FortiAI' empowers you to create robust security playbooks that enha ...
+The AI Assistant Widget adds the **FortiAI** bot to both the list and detailed view of modules in FortiSOAR, as well as in the playbook designer. **FortiAI** is a valuable addition to your security toolkit, making playbook design more efficient and insightful. Whether you are new to security automation or a seasoned professional, **FortiAI** empowers you to create robust security playbooks that e ...
```
**FortiAI** is a valuable addition to your security toolkit, making playbook design more efficient and insightful. --> it more than this like context sensitive conversation

cs-suyog-jinturkar replied:
including learning about security processes --> learning or ML is not present in fortiAI

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,93 +3,101 @@
+![FortiAI dialog](./res/ai_bot_dialog.png)
 
-  >**NOTE**: When a user logs out, all conversations are automatically cleared.
+To manually clear a conversation in the bot, click the **Clear Conversation** button.
```
Note can be added here, the clear conversation option will also delete assistant thread in genai like openai

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,93 +3,101 @@
 
-You can review the generated playbook block by opening specific steps and reviewing their details:   
-![Reviewing an AI generated playbook block](../docs/res/ai_block_reviewstep.png)
+In the playbook designer, you can also ask for information about best practices, jinja, etc., by selecting the **Response Plans / Jinja / Playbooks How Tos** option. The **FortiAI** bot provides insightful responses, leveraging its extensive knowledge base to provide you with guidance, recommendations, or information related to security threats, Jinja, best practices, response plan creation, etc. ...
```
1. selecting the **Response Plans / Jinja / Playbooks How Tos** option. ---> this option is not present now in the bot.
2.  you can also ask for information about best practices, jinja, etc. --> this will be the  of conversation mode here and can be added in use-case section of SP 

In short, above section is not applicable now and only need to consider #2 mentioned above.

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,93 +3,101 @@
   ![Dialog Box](./res/alert_summary_edit_prompt.png)
 
-- Upon clicking 'Submit & Get AI's Response' button the question prompt and module record data will be submitted to the LLM like OpenAI and user will get a response as shown below;
+- Upon clicking the 'Submit & Get AI's Response' button the question prompt and module record data will be submitted to the LLM like OpenAI and the user will get a response as shown below;
```
'Submit & Get AI's Response' --> this button is not present in the bot. please check once.

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,21 +1,10 @@
-    - MITRE ATT&CK Insight
+- The AI Assistant can understand and respond to context sensitive conversations and perform given SOC tasks
+    - Through enhanced comprehension, the bot can navigate to modules, its records, and playbook logs
+    - Sensitive data is being actively masked for all interactions with the API, considering the various modules to which the bot has access
```
sensitive data such as: ['URL', 'Email', 'Domain' , 'Host' , 'IP',  'MD5', 'SHA1', 'SHA256']. Please mention it wherever possible.

but this should not be partt of a release note as its already present in the FortiAI from last version.

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,21 +1,10 @@
+- The AI Assistant can understand and respond to context sensitive conversations and perform given SOC tasks
+    - Through enhanced comprehension, the bot can navigate to modules, its records, and playbook logs
+    - Sensitive data is being actively masked for all interactions with the API, considering the various modules to which the bot has access
+    - The bot can now generate playbooks for all applicable modules and now has access to create and update records
```
The bot can now generate playbooks for all applicable modules --> this is present from previous release. It shouldn't be part of RN

cs-suyog-jinturkar replied:
and now has access to create and update record --> this can be added as usecase and such use-cases cab added too which are supported by FrotiAI as part of conversation

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,21 +1,10 @@
+    - Through enhanced comprehension, the bot can navigate to modules, its records, and playbook logs
+    - Sensitive data is being actively masked for all interactions with the API, considering the various modules to which the bot has access
+    - The bot can now generate playbooks for all applicable modules and now has access to create and update records
+    - The modes &mdash; Playbook Blocks and Conversation Mode &mdash; have been deprecated
```
remove and not deprecated

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,21 +1,10 @@
+    - The bot can now generate playbooks for all applicable modules and now has access to create and update records
+    - The modes &mdash; Playbook Blocks and Conversation Mode &mdash; have been deprecated
+
+- The past conversations remain in memory and help provide context to the bot, even after the user has logged out.
```
not in memory. The past conversations are retained until user clicks on clear conversation. More details can be found in solution pack doc where we have mentioned assistant api details

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,21 +1,10 @@
+    - The modes &mdash; Playbook Blocks and Conversation Mode &mdash; have been deprecated
+
+- The past conversations remain in memory and help provide context to the bot, even after the user has logged out.
+    - These conversations are loaded into the window on login.
```
These conversations are loaded into the window on login. --> this should not be a separate point in RN and can be clubbed in above point.

cs-abhishek-shukla replied:
It is a sub point. Kind of a corollary, expanding on the first line. Shorter sentences help enhance readability. Open to discuss other PoVs as well.

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,21 +1,10 @@
-    - Generate Investigation Report
-    - MITRE ATT&CK Insight
+- The AI Assistant can understand and respond to context sensitive conversations and perform given SOC tasks
+    - Through enhanced comprehension, the bot can navigate to modules, its records, and playbook logs
```
there are more such use-cases like alert investigation, remediation using block ioc, etc.

cs-suyog-jinturkar replied:
escalate alert to an incident, setup warroom, etc.

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,21 +1,10 @@
-    - Generate Response Recommendations
-    - Generate Investigation Report
-    - MITRE ATT&CK Insight
+- The AI Assistant can understand and respond to context sensitive conversations and perform given SOC tasks
```
bot is also cable to to ask you questions if it needs any information/contxt to perform the task like Bloc IOC prompt, bot will ask you IOC value, type, reason, etc.

cs-pranali-chavan reviewed (changes requested):
Looks good.
Rest Suyog has added comments

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,21 +1,10 @@
 # What's New
```
Improved validations like it will check for connector and its configuration is present or not.

cs-abhishek-shukla committed:
3.0.0 changes
- Added release notes and made changes in the doc as per RN
- Removed mention of various modes
- Enhanced release notes to include other features

cs-ankit-gupta commented on README.md:
```diff
@@ -12,18 +12,17 @@
-</table>
-
-**NOTE**: The AI Assistant Widget must be used in conjunction with the 'FortiAI' solution pack to be of any use. For more information, see the [FortiAI solution pack documentation](https://github.com/fortinet-fortisoar/solution-pack-fortinet-advisor/blob/release/2.0.0/README.md).
+>[!NOTE]
```
Version - 3.0.0
Compatibility - 7.6.0 and later

cs-ankit-gupta commented on docs/setup.md:
```diff
@@ -3,21 +3,31 @@
+4. Click **Install** on the lower part of the screen to begin installation.
+
+> [!NOTE]  
+> The **FortiAI Configuration** widget installs with the *FortiAI* solution pack and hence may not require separate installation.
```
FortiAI Configuration - AI Assistant

cs-ankit-gupta commented on docs/usage.md:
```diff
@@ -3,93 +3,104 @@
+- Click on any of the questions, and a dialog box appears to Review and Edit the data that will be sent to the LLM like OpenAI.
+
+>[!NOTE]
+>All sensitive and confidential information is masked automatically by the bot.
```
We need to remove pop on clicking static question

cs-abhishek-shukla committed:
3.0.0 changes
- Reviewed and updated setup.md
- Added SS as part of review
- Removed redundant SS
- Changes made as per review comments

cs-abhishek-shukla committed:
Merge branch 'release/3.0.0' into release/3.0.0

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -3,21 +3,31 @@
+4. Click **Install** on the lower part of the screen to begin installation.
+
+> [!NOTE]  
+> The **FortiAI Configuration** widget installs with the [*FortiAI* solution pack](https://github.com/fortinet-fortisoar/solution-pack-fortinet-advisor/blob/develop/docs/setup.md#installation) and hence may not require separate installation.
```
AI Assistant and not config wizard
